azurerm/internal/services/datamigration/tests/resource_arm_data_migration_service_test.go

```diff
@@ -0,0 +1,210 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMDataMigrationService_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_migration_service", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataMigrationService_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataMigrationServiceExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "virtual_subnet_id"),
+					resource.TestCheckResourceAttr(data.ResourceName, "sku_name", "Standard_1vCores"),
+					resource.TestCheckResourceAttr(data.ResourceName, "kind", "Cloud"),
+					resource.TestCheckResourceAttr(data.ResourceName, "provisioning_state", "Succeeded"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDataMigrationService_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_data_migration_service", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataMigrationServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataMigrationService_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataMigrationServiceExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMDataMigrationService_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_data_migration_service"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMDataMigrationService_complete(t *testing.T) {
```

Comment:
the complete task should not have an update step, there should be a seperate update test